scipy/stats/stats.py

```diff
@@ -5718,3 +5721,195 @@ def rankdata(a, method='average'):
+    verified.
+
+    Intuitively, the method works well if A(c) fills up most of the
+    enclosing rectangle such that the probability is high that (U, V) lies in
+    A(c) whenever it lies in R(c) as the number of required iterations
+    becomes too large otherwise. To be more precise, note that the expected
+    number of iterations to draw (U, V) uniformly distributed on R(c) such that
+    (U, V) is also in A(c) is given by the ratio
+    area(R(c)) / area(A(c)) = 2 * y2 * (x2 - x1), using the fact that
+    the area of A(c) is equal to 1/2 (Theorem 7.1 in [1]_).
+
+    If the bounding rectangle is not correctly specified (i.e. if it does not
+    contain A(c)), the algorithm samples from a distribution different from
+    the one given by the pdf. It is therefore recommended to perform a
+    test such as `stats.kstest` as a check.
+
+    References
+    ----------
+    .. [1] L. Devroye, "Non-Uniform Random Variate Generation",
+       Springer-Verlag, 1986.
+
+    .. [2] W. Hoermann and J. Leydold, "Generating generalized inverse Gaussian
+       random variates", Statistics and Computing, 24(4), p. 547--557, 2014.
+
+    Examples
+    --------
+    >>> from scipy import stats
+
+    Simulate normally distributed random variables. It is easy to compute the
+    bounding rectangle explicitly in that case.
+    >>> f = stats.norm.pdf
+    >>> x_bound = np.sqrt(f(np.sqrt(2))) * np.sqrt(2)
+    >>> x1, x2, y2 = -x_bound, x_bound, np.sqrt(f(0))
+    >>> np.random.seed(12345)
+    >>> rvs = stats.rvs_ratio_unif(f, x1=x1, x2=x2, y2=y2, size=2500)
+
+    The K-S test confirms that the random variates are indeed normally
+    distributed (p-value does not reject normality):
+    >>> stats.kstest(rvs, 'norm')[1]
+    0.373298699196806752
+
+    If the bounding rectangle is not specified, it can be determined
+    numerically over the region specified by `bounds`.
+
+    >>> np.random.seed(12345)
+    >>> rvs = stats.rvs_ratio_unif(f, size=2500, bounds=(-100, 100))
+    >>> stats.kstest(rvs, 'norm')[1]
+    0.3732986992033126
+
+    The exponential distribution provides another example where the bounding
+    rectangle can be determined explicitly.
+    >>> np.random.seed(12345)
+    >>> rvs = stats.rvs_ratio_unif(lambda x: np.exp(-x), x1=0, x2=2*np.exp(-1),
+    ...                            y2=1, size=1000)
+    >>> stats.kstest(rvs, 'expon')[1]
+    0.52369231518316373
+
+    Sometimes it can be helpful to use a non-zero shift parameter, see e.g.
+    [2]_ above in the case of the generalized inverse Gaussian distribution.
+
+    """
+
+    def find_rect_bounds(g, bounds, var):
+        if bounds is None:
+            raise Exception("The parameter bounds needs to be specified if " +
+                            var + " is None.")
+        opt_res = optimize.minimize_scalar(g, bounds=bounds, method='Bounded')
+        if not opt_res.success:
+            raise Exception("Optimization failed when attempting to find " +
```

Comment:
You usually never raise an `Exception`, but something more specific. `RuntimeError` is probably a good choice here.

And same above and below.